invenio_rdm_records/collections/models.py

```diff
@@ -156,9 +156,30 @@ def get_by_slug(cls, slug, tree_id):
         return cls.query.filter(cls.slug == slug, cls.tree_id == tree_id).one_or_none()
 
     @classmethod
-    def read_many(cls, ids):
+    def read_many(cls, ids_):
         """Get many collections by ID."""
-        return cls.query.filter(cls.id.in_(ids)).order_by(cls.path, cls.order)
+        return cls.query.filter(cls.id.in_(ids_)).order_by(cls.path, cls.order)
+
+    @classmethod
+    def read_all(cls):
+        """Get all collections."""
+        return cls.query.order_by(cls.path, cls.order)
+
+    def update(
+        self, /, slug=None, title=None, search_query=None, order=None, num_records=None
```

Comment:
My idea with this is to make the update very explicit, to avoid things being updated without the developer's knowledge. 

Moreover, I don't think some fields should be updated (e.g. `id`, `depth` or `path` since those are computed).

Reply:
What about (pseudocde):
```python
    def update(self, **kwargs):
        """A very nice docstring to explain what key/values are expected."""
        allowed_keys = ["slug", ...]
        for key, value in kwargs.items():
          if key in allowed_keys:
            setattr(self, key, value)
```